gridt/src/app/core/settings.service.ts

```diff
@@ -0,0 +1,44 @@
+import { Injectable } from "@angular/core";
+import { Observable, pipe, throwError, of, iif } from "rxjs";
+import { flatMap, tap, map, catchError, pluck, mergeMap, take } from "rxjs/operators";
+
+import { Identity } from './models/identity.model';
+import { AuthService } from './auth.service';
+import { SecureStorageService } from './secure-storage.service';
+
+@Injectable({
+  providedIn: "root"
+})
+
+export class SettingsService {
+
+  constructor (
+    private auth: AuthService,
+    private secStore: SecureStorageService,
+  ) {}
+
+  /**
+   * Observable to obtain settings from secure storage
+   * If secure storage is empty, this returns an error
+   */
+  public localIdentity$: Observable<Identity> = this.auth.readyAuthentication$.pipe(
+    flatMap(() => this.secStore.get$("identity"))
+  );
+
+  /**
+   * Store settings in the secure storage.
+   * @param identity Settings to be stored.
+   */
+  public setLocalIdentity$(identity: Identity): Observable<boolean> {
+    return this.auth.isLoggedIn$.pipe(
+      take(1),
+      tap( (val) => {
+        if (val === true) {
+          console.debug(`Storing identity: ${JSON.stringify(identity)} into the local storage.`);
+          this.secStore.set$("identity", identity).subscribe();
+        }
+      }),
+      mergeMap( val => iif( () => val === true, of(true), throwError("Not logged in: can't store identity in local storage.")))
```

Comment:
This should be a mapTo

https://www.learnrxjs.io/learn-rxjs/operators/transformation/mapto

And a catchError/throwError thing

Reply:
I don't see why you'd want this to map to a single value. The whole idea of using the `iff()` is to return `of(true)` when successful and and error when not.

Reply:
If there is an error, mapTo should ignore it and just pass on the error. The error you emit is different from the one you get from secstore so a catchError/throwError can be used to perform that replacement.

From the filter you know you will only obtain values that are true, so this observable will always emit true, unless it emits an observable, mapTo makes that very clear.